ml-fundamentals/ml-project-management.md

```diff
@@ -0,0 +1,46 @@
+# Agile development considerations for ML projects
+
+## Goals
+
+1. Run and manage ML projects effectively
+2. Create effective collaboration between the ML team and the other teams working on the project
+
+## Agile project management
+
+When running ML projects, we follow the Agile methodology for software development with some adaptations, as we acknowledge that research and experimentation are sometimes difficult to plan and estimate.
+
+To learn more about how CSE runs the Agile process for software development teams, refer to this [doc](../agile-development).
+
+Within this framework, the team follows these Agile ceremonies:
+
+- [Backlog management](../agile-development/backlog-management/readme.md)
+- [Retrospectives](../agile-development/retrospectives/readme.md)
+- [Scrum of Scrums](../agile-development/scrum-of-scrums/readme.md) (where applicable)
+- [Sprint planning](../agile-development/sprint-planning/readme.md)
+- [Standups](../agile-development/stand-ups/readme.md)
+- [Working agreement](../agile-development/team-agreements/working-agreements/readme.md)
+
+### Notes on Agile process during exploration and experimentation
```

Comment:
In CSE projects, we strive to get to an end to end "hello world" project repo with automated testing and DevOps ASAP so we can focus on iterating on different workstreams independently. 

What is that for ML? How can I get all that set up quickly so it's easy for the DS/ML team to start creating notebooks/scripts/models quickly while still working on the rest of the system (data collection, the rest of the solution around a placeholder/interim model, etc.).

Reply:
I'm not sure we have a one-size-fits-all here. For the feasibility study a CI process would likely be an overkill. After we finished with feasibility study, the ml-experimentation doc describes how to set up the env and code to get started. However some people would still not want things like linting for example at this stage. Let's continue the discussion and see if we can add a general guidance here.